_posts/2020-03-02-tomcat-thread-model.md

```diff
@@ -0,0 +1,96 @@
+
+Co więcej zauważyliśmy i potwierdziliśmy to później w testach, że po całkowitym wysyceniu puli wątków Tomcat nie jest w stanie bez restartu powrócić do prawidłowego działania nawet wtedy, kiedy problem z długimi czasami odpowiedzi zostanie wyeliminowany.
+
+Opisana sytuacja skłoniła nas do sprawdzenia czy jesteśmy w stanie zapobiec takiemu zachowaniu Tomcata poprzez modyfikację konfiguracji connectorów. Pojawił się między innymi pomysł zmiany implementacji connectora blokującego (bio) na nieblokujący (nio). W systemie, którego dotyczył problem używany był tomcat 7.x, w którym domyślnie używana jest implementacja blokująca.
+
+## Konfiguracja connectorów
+
+Podstawowa konfiguracja Tomcata znajduje się w pliku conf/server.xml i zawiera konfigurację następujących elementów:
+* [Server](http://tomcat.apache.org/tomcat-7.0-doc/config/server.html) - konfiguracja całego kontenera, zawiera konfiguracje poszczególnych serwisów.
+* [Service](http://tomcat.apache.org/tomcat-7.0-doc/config/service.html) - Zawiera konfigurację poszczególnych connectorów łącząc je silnikiem przetwarzania żądań.
+* [Connector](http://tomcat.apache.org/tomcat-7.0-doc/config/http.html) - nasłuchuje na wybranym porcie http i obsługuje połączenia przekazując żądania do silnika zdefiniowanego w elemencie Engine.
+* [Engine](http://tomcat.apache.org/tomcat-7.0-doc/config/engine.html) - przetwarza żądania pochodzące ze wszystkich connectorów zdefiniowanych w elemencie Service.
+
+Z punktu widzenia przetwarzania żądań kluczowe znaczenie ma konfiguracja connectora i parametry:
+* protocol - pozwala określić wybraną implementacje connectora (to tutaj możemy zdecydować czy chcemy użyć connectora blokującego czy nieblokującego).
+* acceptCount - długość kolejki oczekujących połączeń. Kolejka napełnia się jeżeli wszystkie wątki w puli wątków connectora są zajęte. W przypadku osiągnięcia limitu tej kolejki (domyślnie 100) połączenia klientów będą odrzucane.
+* maxConnections - maksymalna liczba połączeń, które mogą być przetwarzane. Po osiągnięciu maksymalnej liczby połączeń, połączenia nadal są przyjmowane i kolejkowane do czasu osiągnięcia limitu wynikającego z parametru acceptCount. Domyślnie maxConnections jest równe maxThreads dla connectora blokującego i 10000 dla nieblokującego.
+* maxThreads - maksymalna liczba wątków obsługujących żądania. Parametr ten oznacza ile żądań może być symultanicznie przetwarzanych przez serwer. Domyślna wartość tego parametru to 200. Biorąc pod uwagę problem, którym zajmujemy się w tym artykule jest to najważniejszy parametr.
```

Comment:
zamiast kropek na końcu wyliczania, powinny być średniki
https://www.ekorekta24.pl/interpunkcja-w-wyliczeniach-przecinki-sredniki-a-moze-nic/

Reply:
przecinek:
Parametr ten oznacza ile żądań może być symultanicznie przetwarzanych przez serwer. Domyślna wartość tego parametru to 200. ==> Parametr ten oznacza, ile żądań może być symultanicznie przetwarzanych przez serwer. Domyślna wartość tego parametru to 200. 

A tutaj daję na sugestię drobną inwersję, mi się wydaje że lepiej działa słownomuzycznie:
Biorąc pod uwagę problem, którym zajmujemy się w tym artykule jest to najważniejszy parametr.
==>
Jest to najważniejszy parametr, zważywszy na problem, który rozwiązujemy w tym artykule.

Reply:
Co do tych średników to się nie zgodzę. Zamieniłem wszędzie na zdania. Bo tam już były zdania. To by źle wyglądało. Nagle na końcu jednego zdania średnik zamiast kropki?

Przecinek i sugestię przyjmuję.